openstef/tasks/calculate_kpi.py

```diff
@@ -40,26 +40,56 @@
 THRESHOLD_OPTIMIZING = 0.50
 
 
-def main(model_type=None):
+def main(model_type: MLModelType = None, config=None, database=None) -> None:
     taskname = Path(__file__).name.replace(".py", "")
+
+    if database is None or config is None:
+        raise RuntimeError(
+            "Please specifiy a configmanager and/or database connection object. These can be found in the openstef-dbc package."
+        )
+
     if model_type is None:
         model_type = [ml.value for ml in MLModelType]
 
-    with TaskContext(taskname) as context:
+    with TaskContext(taskname, config, database) as context:
         # Set start and end time
         start_time = datetime.utcnow() - timedelta(days=1)
         end_time = datetime.utcnow()
 
         PredictionJobLoop(context, model_type=model_type).map(
-            check_kpi_pj,
+            check_kpi_task,
             context,
             start_time=start_time,
             end_time=end_time,
         )
 
 
-def check_kpi_pj(pj, context, start_time, end_time):
-    kpis = calc_kpi_for_specific_pid(pj["id"], start_time=start_time, end_time=end_time)
+def check_kpi_task(
+    pj: PredictionJobDataClass,
+    context: TaskContext,
+    start_time: datetime,
+    end_time: datetime,
+) -> None:
+    # Apply default parameters if none are provided
+    if start_time is None:
+        start_time = datetime.utcnow() - timedelta(days=1)
+    if end_time is None:
+        end_time = datetime.utcnow()
+
+    # Get realised load data
+    realised = context.database.get_load_pid(pj["id"], start_time, end_time, "15T")
```

Comment:
Do we we want to keep this database logic in openstef? Maybe for now it's okay, but it would be nice to move all database logic to the application above and only keep forecasting logic in openstef. If you agree, you can add a todo or an issue/user story to change this later :)

Reply:
well spotted, I did not realize this was still here, I will remove it.